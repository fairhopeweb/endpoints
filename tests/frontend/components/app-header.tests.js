import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import AppHeader from '@/components/app-header.vue';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);

describe('AppHeader', () => {
    let state;
    let getters;
    let actions;
    let store;

    beforeEach(() => {
        state = {
            hash: 'abc123',
            isPerformingActions: false
        };
        getters = {
            hasAvailableEndpoints: () => false
        };
        actions = {
            createEndpoint: jest.fn()
        };
        store = new Vuex.Store({ state, getters, actions });
    });

    test('is a Vue instance', () => {
        const wrapper = shallowMount(AppHeader, { store, localVue });
        expect(wrapper.isVueInstance()).toBeTruthy();
    });

    test('matches snapshot', () => {
        const wrapper = shallowMount(AppHeader, { store, localVue });
        expect(wrapper.element).toMatchSnapshot();
    });
});
