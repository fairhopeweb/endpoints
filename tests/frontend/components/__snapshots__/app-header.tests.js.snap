// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader matches snapshot 1`] = `
<header
  class="app-header"
>
  <navigation-loader-stub />
   
  <div
    class="wrapper"
  >
    <router-link-stub
      class="logo"
      event="click"
      tag="a"
      to="[object Object]"
    >
      Endpoints
    </router-link-stub>
     
    <base-button-stub
      class="primary"
      type="button"
    >
      Create Endpoint
    </base-button-stub>
     
    <!---->
  </div>
</header>
`;
